validator/server_arb/execution_run.go

```diff
@@ -79,6 +83,119 @@ func (e *executionRun) GetStepAt(position uint64) containers.PromiseInterface[*v
 	})
 }
 
+func (e *executionRun) GetMachineHashesWithStepSize(fromBatch, machineStartIndex, stepSize, numDesiredLeaves uint64) containers.PromiseInterface[[]common.Hash] {
+	return stopwaiter.LaunchPromiseThread(e, func(ctx context.Context) ([]common.Hash, error) {
+		return e.machineHashesWithStepSize(ctx, machineHashesWithStepSizeArgs{
+			startIndex:        machineStartIndex,
+			fromBatch:         fromBatch,
+			stepSize:          stepSize,
+			requiredNumHashes: numDesiredLeaves,
+		})
+	})
+}
+
+type GlobalStateGetter interface {
+	GetGlobalState() validator.GoGlobalState
+	HashStepper
+}
+
+type HashStepper interface {
```

Comment:
We already have defined a MachineInterface which is implemented by both *ArbitratorMachine and *IncorrectMachine.. I think it's better to use that interface and also implement it for MockMachine, to keep complexity/num of interfaces lower for business logic even at the expense of a little more testing logic.

In that cases MachneHashesWithStepSize will probably get a MachineInterface (already at first step), stepSize, requiredNumHashes